judyz_cffi/_common.py

```diff
@@ -8,7 +8,7 @@
 
 
 class _JudyCommon:
-    def update(self, other: Optional[Union[Mapping[TKey, int], Iterable[Tuple[TKey, int]]]]) -> None:
+    def update(self, other: Mapping[TKey, int] | Iterable[tuple[TKey, int]] | None) -> None:
```

Comment:
https://peps.python.org/pep-0585/ is only on Python 3.9+ BUT `from __future__ import annotations` means we don't care. 

Mypy will recognize those even when running on earlier versions of Python. The only thing to be aware of is that if we need to use annotations at runtime, this could be complicated, but I think it should be alright :+1: